miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/ChatMessageBridgeDispatcher.kt

```diff
@@ -0,0 +1,71 @@
+package com.rakuten.tech.mobile.miniapp.js
+
+import com.google.gson.Gson
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionCache
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionType
+
+/**
+ * Functionality related to Chat/Messaging. These methods should be implemented by the Host App.
+ */
+@Suppress("TooGenericExceptionCaught")
+abstract class ChatMessageBridgeDispatcher {
+
+    private lateinit var bridgeExecutor: MiniAppBridgeExecutor
+    private lateinit var customPermissionCache: MiniAppCustomPermissionCache
+    private lateinit var miniAppId: String
+
+    /**
+     * Triggered when Mini App wants to send a message to a contact.
+     * Should open a contact chooser which allows the user to choose a single contact,
+     * and should then send the message to the chosen contact.
+     * Should invoke [onSuccess] with the ID of the contact which was sent the message.
+     * If the user cancelled sending the message, should invoked [onSuccess] with [null].
+     * Should invoke [onError] when there was an error.
+     */
+    abstract fun sendMessageToContact(
+        message: MessageToContact,
+        onSuccess: (contactId: String?) -> Unit,
+        onError: (message: String) -> Unit
+    )
+
+    internal fun init(
+        bridgeExecutor: MiniAppBridgeExecutor,
+        miniAppCustomPermissionCache: MiniAppCustomPermissionCache,
+        miniAppId: String
+    ) {
+        this.bridgeExecutor = bridgeExecutor
+        this.customPermissionCache = miniAppCustomPermissionCache
+        this.miniAppId = miniAppId
+    }
+
+    internal fun onSendMessageToContact(callbackId: String, jsonStr: String) = try {
+        if (customPermissionCache.hasPermission(
```

Comment:
@corycaywood @khairul-alam-licon I think the sdk does not need to check the contact permission here when miniapp wants to send message. Hostapps can control this on their own.

Reply:
Yeah, I think you're right @minh-rakuten . The contact permission is specifically for the `getContacts` feature, so this shouldn't be checked here (or the host app can check it if they want to).

Reply:
Thank you for sharing, Yeah, we can skip the permission checking here. I'll update this part.